api/app/auth.py

```diff
@@ -13,19 +15,38 @@
 
 
 async def authenticate(token: str = Depends(oauth2_scheme)):
-    """ Returns True when validation of the token is successful """
+    """ Returns True when validation of the token is successful. """
     # RSA public key format
     keycloak_public_key = '-----BEGIN PUBLIC KEY-----\n' + \
         config.get('KEYCLOAK_PUBLIC_KEY') + '\n-----END PUBLIC KEY-----'
 
     try:
-        jwt.decode(token, keycloak_public_key, algorithm='RS256')
-        return True
-    except Exception as exception:
-        detail = 'Could not validate the credential ({})'.format(exception)
+        decoded_token = jwt.decode(token, keycloak_public_key, algorithm='RS256')
+        return decoded_token
+    except InvalidTokenError:
+        detail = 'Could not validate the credential (Not enough segments)'
```

Comment:
Rather log the exception - we don't know that InvalidTokenError always means not enough segments, it could be something else.
```suggestion
        logger.error('Could not validate the credential', exception)
```